requirements.txt

```diff
@@ -0,0 +1,10 @@
+# required dependencies
+pyside6
+pygraphviz
+
+# optional dependencies
+stringtemplate3
+singledispatch
```

Comment:
Why optional? `singledispatch` is required.

Reply:
[This code](https://github.com/esa/opengeode/blob/master/opengeode/StgBackend.py#L26) tells me the `singledispatch` *decorator* is required, but the `singledispatch` *package* is needed only for `python2` ([backport](https://pypi.org/project/singledispatch/)).
With python3, `singledispatch` is imported from the package `functools`.

Also [python documentation](https://docs.python.org/3/library/functools.html#functools.singledispatch) says that the singledispatch decorator is part of `functools` since version 3.4.

Should we care about supporting python versions <=3.4 ?
TASTE is using 3.5 from what I can recall, and the `setup.py` [says >=3.7](https://github.com/esa/opengeode/blob/master/setup.py#L27) ...

Reply:
Oh so the `singledispatch` package is unnecessary now and indeed this dependency could be removed (and replaced with an import from `functools`). I missed that, thanks for spotting it.